steps/metabase/base/metabase.go

```diff
@@ -0,0 +1,88 @@
+package base
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	gabs "github.com/Jeffail/gabs/v2"
+	"github.com/stackpulse/steps-sdk-go/env"
+)
+
+type MetabaseCommandArgs struct {
+	Username  string `env:"USERNAME,required"`
+	Password  string `env:"PASSWORD,required"`
+	Host      string `env:"HOST,required" envDefault:"metabase.data.svc.cluster.local"`
+	Token     string `env:"TOKEN," envDefault:""`
+	Recipient string `env:"RECIPIENT"`
+	UserId    string
+}
+
+type Output struct {
+	User    string
+	Message string
+}
+
+type MetabaseCommand struct {
+	MetabaseCommandArgs
+}
+
+func NewMetabaseCommand(args *MetabaseCommandArgs) (*MetabaseCommand, error) {
+	if err := env.Parse(args); err != nil {
+		return nil, fmt.Errorf("parse args: %w", err)
+	}
+	return &MetabaseCommand{MetabaseCommandArgs: *args}, nil
+}
+
+func (mb *MetabaseCommand) GetUserId() (*MetabaseCommand, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/api/user/", mb.Host), nil)
+	req.Header.Add("X-Metabase-Session", mb.Token)
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("Http error: %w", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Http error: %w", err)
+	}
+	users, err := gabs.ParseJSON([]byte(body))
+	if err != nil {
+		return nil, fmt.Errorf("json parse error: %w", err)
+	}
+	fmt.Println("User list %w", users)
+	for _, user := range users.Children() {
+		if user.S("email").Data().(string) == mb.Recipient {
+			mb.UserId = fmt.Sprintf("%g", user.S("id").Data().(float64))
+			break
+		}
+	}
+	return mb, nil
+}
+
+func (mb *MetabaseCommand) GetSessionToken() (*MetabaseCommand, error) {
+	// var token string
+	postBody, _ := json.Marshal(map[string]string{
+		"username": mb.Username,
+		"password": mb.Password,
+	})
+	requestBody := bytes.NewBuffer(postBody)
+	resp, err := http.Post(fmt.Sprintf("%s/api/session", mb.Host), "application/json", requestBody)
+	if err != nil {
+		return nil, fmt.Errorf("Http error: %w", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Http error: %w", err)
+	}
+	token, err := gabs.ParseJSON([]byte(body))
+	if err != nil {
+		return nil, fmt.Errorf("json parse error: %w", err)
+	}
+	mb.Token = token.S("id").Data().(string)
```

Comment:
check that the conversion succeeded
```suggestion
	s, ok := v.(string)
        if !ok {
            // the assertion failed.
        }
```